packages/twenty-front/src/pages/settings/data-model/SettingsObjectDetailPage.tsx

```diff
@@ -83,13 +87,33 @@ export const SettingsObjectDetailPage = () => {
     findActiveObjectMetadataItemBySlug(updatedObjectSlug);
 
   const { activeTabIdState } = useTabList(TAB_LIST_COMPONENT_ID);
+  const { setActiveTabId } = useTabList(TAB_LIST_COMPONENT_ID);
   const activeTabId = useRecoilValue(activeTabIdState);
 
   const isAdvancedModeEnabled = useRecoilValue(isAdvancedModeEnabledState);
   const isUniqueIndexesEnabled = useIsFeatureEnabled(
     'IS_UNIQUE_INDEXES_ENABLED',
   );
 
+  const hash = location.hash.replace('#', '');
+
+  useEffect(() => {
+    if (!hash) {
+      const initialTabId = activeTabId ?? FIELDS_TAB_ID;
+      navigate(`${location.pathname}#${initialTabId}`, { replace: true });
```

Comment:
aaargh! Can we avoid useEffects? We should update the location when the user actually navigates / clicks